tools/scalajsenv.js

```diff
@@ -968,40 +992,40 @@ ScalaJS.TypeData.prototype["newArrayOfThisClass"] = function(lengths) {
 
 // Create primitive types
 
-ScalaJS.d.V = new ScalaJS.TypeData().initPrim(undefined, "V", "void");
-ScalaJS.d.Z = new ScalaJS.TypeData().initPrim(false, "Z", "boolean");
-ScalaJS.d.C = new ScalaJS.TypeData().initPrim(0, "C", "char");
-ScalaJS.d.B = new ScalaJS.TypeData().initPrim(0, "B", "byte");
-ScalaJS.d.S = new ScalaJS.TypeData().initPrim(0, "S", "short");
-ScalaJS.d.I = new ScalaJS.TypeData().initPrim(0, "I", "int");
-ScalaJS.d.J = new ScalaJS.TypeData().initPrim("longZero", "J", "long");
-ScalaJS.d.F = new ScalaJS.TypeData().initPrim(0.0, "F", "float");
-ScalaJS.d.D = new ScalaJS.TypeData().initPrim(0.0, "D", "double");
+ScalaJS.d.V = new ScalaJS.TypeData().initPrim(undefined, "V", "void", 0);
+ScalaJS.d.Z = new ScalaJS.TypeData().initPrim(false, "Z", "boolean", 1);
+ScalaJS.d.C = new ScalaJS.TypeData().initPrim(0, "C", "char", 2);
+ScalaJS.d.B = new ScalaJS.TypeData().initPrim(0, "B", "byte", 3);
+ScalaJS.d.S = new ScalaJS.TypeData().initPrim(0, "S", "short", 4);
+ScalaJS.d.I = new ScalaJS.TypeData().initPrim(0, "I", "int", 5);
+ScalaJS.d.J = new ScalaJS.TypeData().initPrim("longZero", "J", "long", 6);
+ScalaJS.d.F = new ScalaJS.TypeData().initPrim(0.0, "F", "float", 7);
+ScalaJS.d.D = new ScalaJS.TypeData().initPrim(0.0, "D", "double", 8);
 
 // Instance tests for array of primitives
 
-ScalaJS.isArrayOf.Z = ScalaJS.makeIsArrayOfPrimitive(ScalaJS.d.Z);
+ScalaJS.isArrayOf.Z = ScalaJS.makeIsArrayOfPrimitive(ScalaJS.d.Z.typeTag);
```

Comment:
Consider hard-coding the constants here. Hopefully this could help GCC to dce away `ScalaJS.d.V` (and similar) if the data are not accessed by the `isArrayOf`s are needed.